homeassistant/components/lock/homekit_controller.py

```diff
@@ -0,0 +1,112 @@
+"""
+Support for HomeKit Controller locks.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/lock.homekit_controller/
+"""
+
+import logging
+
+from homeassistant.components.homekit_controller import (HomeKitEntity,
+                                                         KNOWN_ACCESSORIES)
+from homeassistant.components.lock import LockDevice
+from homeassistant.const import (STATE_LOCKED, STATE_UNLOCKED,
+                                 STATE_UNKNOWN, ATTR_BATTERY_LEVEL)
+
+DEPENDENCIES = ['homekit_controller']
+
+_LOGGER = logging.getLogger(__name__)
+
+STATE_JAMMED = 'jammed'
```

Comment:
Only base lock component states, in this case defined in const.py, are allowed. New states would require approval in an issue in our architecture repo.

Reply:
Sorry, I realized that it doesn't matter for the lock state what other states we populate this dict with, since we only check if the internal state is locked or not in `is_locked`. It's ok to define a constant `STATE_JAMMED` in this module as long as we don't actually return that state as entity state, but follow the base lock entity implementation, which we do here.